rabbitmq/check.py

```diff
@@ -293,6 +299,29 @@ def _get_metrics(self, data, object_type):
                     self.log.debug("Caught ValueError for %s %s = %s  with tags: %s" % (
                         METRIC_SUFFIX[object_type], attribute, value, tags))
 
+    def get_connections_stat(self, instance, base_url, object_type, vhosts=None, auth=None, ssl_verify=True, skip_proxy=False):
+        """
+        Collect metrics on currently open connection per vhost.
+        """
+        instance_proxy = self.get_instance_proxy(instance, base_url)
+        data = self._get_data(urlparse.urljoin(base_url, object_type), auth=auth,
+                              ssl_verify=ssl_verify, proxies=instance_proxy)
+
+        stats = defaultdict(int)
+        for conn in data:
+            if vhosts and conn['vhost'] in vhosts:
+                stats[conn['vhost']] += 1
+            elif not vhosts:
+                stats[conn['vhost']] += 1
+
+        # avoid having no data if no client it connected
+        if len(stats) == 0:
+            self.gauge('rabbitmq.connections', 0, tags=[])
```

Comment:
if `vhosts` is provided, instead of sending `0` with an empty tag list, we should rather send `0` with a vhost tag on every `vhost` that's configured but on which no connection is found. This would allow monitors configured to alert on every `vhost` to not alert on no data when no client is connected on a particular vhost. Does that make sense?

Also, more generally, it's not very useful to send a `0` with a different tag list, so when `vhosts` is not provided we could either remove that `0` altogether, or grab a list of all the vhosts like what the `_check_aliveness` function does